src/db/read.rs

```diff
@@ -36,6 +35,28 @@ impl<'a> OwnedRead<'a> {
     }
 }
 
```

Comment:
can you comment the return types? (also: we should consider having `type Hash = String` and `type HeadName = String` for clarity).
```
// read_commit reads a commit specified by whence from the dag and returns its hash, Commit, and Map.
```